test/UnitTest/Microsoft.AspNet.OData.Test.Shared/ODataStreamPropertyTest.cs

```diff
@@ -0,0 +1,222 @@
+                config.MapODataServiceRoute("odata", "odata", GetEdmModel());
+            });
+            var client = TestServerFactory.CreateClient(server);
+
+            // Act
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, RequestUri);
+            HttpResponseMessage response = await client.SendAsync(request);
+
+            // Assert
+            Assert.True(response.IsSuccessStatusCode);
+
+            Assert.Equal("application/octet-stream", response.Content.Headers.ContentType.MediaType);
+            var stream = await response.Content.ReadAsStreamAsync();
+
+            StreamReader reader = new StreamReader(stream);
+            string text = reader.ReadToEnd();
+            Assert.Equal("\u0003\u0004\u0005\u0006", text);
+
+            byte[] byteArray = stream.ReadAllBytes();
+            Assert.Equal(new byte[] { 3, 4, 5, 6 }, byteArray);
+        }
+
+        private static IEdmModel GetEdmModel()
+        {
+            ODataModelBuilder builder = ODataConventionModelBuilderFactory.Create();
+            builder.EntitySet<StreamCustomer>("StreamCustomers");
+            return builder.GetEdmModel();
+        }
+    }
+
+    public static class StreamExtensions
+    {
+        public static byte[] ReadAllBytes(this Stream instream)
+        {
+            if (instream is MemoryStream)
+                return ((MemoryStream)instream).ToArray();
+
+            using (var memoryStream = new MemoryStream())
+            {
+                instream.CopyTo(memoryStream);
+                return memoryStream.ToArray();
+            }
+        }
+    }
+
+    // Controller
+    public class StreamCustomersController : TestODataController
+    {
+        [EnableQuery]
+        public IQueryable<StreamCustomer> Get()
+        {
+            return CreateCustomers().AsQueryable();
+        }
+
+        public ITestActionResult Get(int key)
+        {
+            IList<StreamCustomer> customers = CreateCustomers();
+            StreamCustomer customer = customers.FirstOrDefault(c => c.Id == key);
+            if (customer == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(customer);
+        }
+
+        [HttpGet]
+        public ITestActionResult GetName(int key)
+        {
+            IList<StreamCustomer> customers = CreateCustomers();
+            StreamCustomer customer = customers.FirstOrDefault(c => c.Id == key);
+            if (customer == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(customer.Name);
+        }
+
+        [HttpGet]
+        public ITestActionResult GetPhoto(int key)
+        {
+            IList<StreamCustomer> customers = CreateCustomers();
+            StreamCustomer customer = customers.FirstOrDefault(c => c.Id == key);
+            if (customer == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(customer.Photo);
+        }
```

Comment:
We need a way to return a media-type other than application/octet-stream here (i.e., image/jpeg)